frontend/islands/vite.config.mts

```diff
@@ -3,10 +3,8 @@
 import fs from "node:fs";
 import path from "node:path";
 import react from "@vitejs/plugin-react";
-import { defineConfig, type Plugin } from "vite";
-import topLevelAwait from "vite-plugin-top-level-await";
+import { defineConfig, type Plugin } from "rolldown-vite";
```

Comment:
Since rolldown-vite will eventually be deprecated in an upcoming release of `vite`, you can do:

```json
  "dependencies": {
    "vite": "npm:rolldown-vite@latest",
  }
```

In the `package.json`. That will lock the specific version of `rolldown-vite` (in our pnpm-lock.yml), but keep `vite` as an alias. That way we can keep imports to:

```ts
import { defineConfig, type Plugin } from "vite";
```

and also run `pnpm vite` to run the rolldown-vite cli.